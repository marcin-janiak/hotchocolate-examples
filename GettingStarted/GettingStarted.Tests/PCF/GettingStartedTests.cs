using System;
using System.IO;
using System.Threading.Tasks;
using GettingStarted.PCF;
using GettingStarted.PCF.Data;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace GettingStarted.Tests.PCF
{
    public class GettingStartedTests
    {
        [Fact]
        public async Task Ensure_Schema_IsCorrect()
        {
            ISchema schema = SchemaBuilder.New().AddQueryType<Query>()
                .Create();

            string schemaSDL = schema.ToString();

            schemaSDL.MatchSnapshot();
        }

        [Fact]
        public async Task GetAuthors_AuthorsAreReturned()
        {
            IServiceProvider serviceProvider =
                new ServiceCollection()
                    .AddDbContext<BooksContext>(options =>
                        options.UseInMemoryDatabase(databaseName: "Books")
                            .UseLazyLoadingProxies())
                    .BuildServiceProvider();

            await serviceProvider.GetService<BooksContext>().Database.EnsureCreatedAsync();
            
            var fileName = "./.Queries/GetAuthors.graphql";
            var query = await File.ReadAllTextAsync($@"{Directory.GetCurrentDirectory()}/{fileName}");

            IQueryExecutor executor = SchemaBuilder.New()
                .AddQueryType<Query>()
                .Create()
                .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                    .SetQuery(query)
                    .SetServices(serviceProvider)
                    .Create();
            
            IExecutionResult result = await executor
                .ExecuteAsync(request);

            result.MatchSnapshot();
        }
    }
}