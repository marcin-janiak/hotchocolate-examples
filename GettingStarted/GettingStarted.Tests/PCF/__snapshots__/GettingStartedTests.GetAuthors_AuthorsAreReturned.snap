{
  "Data": {
    "authors": [
      {
        "firstName": "Andrzej",
        "lastName": "Sapkowski"
      },
      {
        "firstName": "Stephen",
        "lastName": "King"
      }
    ]
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null
}
